ex1/functions.py

```diff
@@ -91,25 +294,165 @@ def f():
 # number of arguments
+def args_to_dict_general(*args):
+    """ Creates a dictionary with any number of arguments, keys: index, value: arguments
+    	
+    Parameters
+    ----------
+    args: any type
+    
+    Returns
+    -------
+    Dictionary
+    
+    See Also
+    --------
+    args_to_dict(a,b,c)
+    
+    Example
+    -------
+    >>>print(args_to_dict_general(1,'22',4.6,'ttt'))
+    {0:1,1:'22',2:4.6,3:'ttt'}
+    """
+    #return dict(zip(range(0,len(args)),args))
+    return {i:j for i,j in enumerate(args)}
 
 # Define a function named lists_to_dict that takes two lists of equal lenght
 # named keys and values and builds a dictionary out of them.
 
+def lists_to_dict(keys,values):
+    """ Creates a dictionary out of two lists with same length
+    	
+    Parameters
+    ----------
+    keys,values: any type
+		Have to have same length
+    
+    Returns
+    -------
+    Dictionary
+    
+    See Also
+    --------
+    args_to_dict(a,b,c)
+    args_to_dict_general(*args)
+    
+    Example
+    -------
+    >>>print(lists_to_dict([0,'aa',3],['22','d',2]))
+    {0:'22','aa':'d',3:2}
+    """
+    return dict(zip(keys,values))
+
 # Define a function named search_list that takes two lists a and b. The
 # function searches for all elements of b in list a. The return value should be
 # a dictionary containing the index of the found element of b in list a and the
 # value of the found element. If nothing was found then return an empty
 # dictionary.
 
+def search_list(a,b):
+    """ Creates a dictionary with elements which are in list a and list b, key: index of the element in list a
+    	
+    Parameters
+    ----------
+    a,b: list
+		element: any type
+    
+    Returns
+    -------
+    Dictionary
+    
+    Example
+    -------
+    >>>print(search_list([1,2,3,4,5,6,7,8],[1,4]))
+    {0:1,3:4}
+    """
+    r_dict = {}
+    for idx,i in enumerate(a):
+        for j in b:
+            if  i == j:
+                r_dict[idx] = i
+    return r_dict
+
 # Define a function named dict_to_string that takes a dictionary and a
 # separator string. The function should only take elements out of the
 # dictionary whose value is a string and then return a single string containing
 # the strings stored in the dictionary seperated by the separator string.
 # Return an empty string if there are no strings in the dictionary.
 
+def dict_to_string(adict,sep):
+    """ Creates a string out of the values in dictionary adict which are strings, this elements will be seperated by sep
+    	
+    Parameters
+    ----------
+    adict: dictionary
+    sep: string
+    
+    Returns
+    -------
+    String
+    
+    Example
+    -------
+    >>>print(dict_to_string({0:'hi','a':1,'c':'ho'},'AND'))
+    'hiANDho'
+    """
+    s = ''
+    for value in adict.values():
+        if isinstance(value, str):
+            if s == '':
+                s = value
+            else: 
+                s = s+sep+value
+    return s
```

Comment:
The `join` method makes this easier. See https://github.com/TUW-GEO-python-intro/Exercise-1/pull/20